### PR TITLE
[docs] remove aliyun from cluster launcher docs

### PR DESCRIPTION
**Pull request by AmeerHajAli**
I don't think aliyun is supported.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by simon-mo** [comment]
or community managed? there hasn't been update since the initial contribution indeed https://github.com/ray-project/ray/commits/master/python/ray/autoscaler/_private/aliyun

**Comment by DmitriGekhtman** [comment]
Recently, we've been asking developers of external custom node providers to host their node providers independently (i.e. not in the Ray repo).
However, we made that policy shift after AliyunNodeProvider was contributed.

If we wanted to remove this node provider from the Ray docs, we would first want to ask the maintainers to move the AliyunNodeProvider code and documentation to another repo.
For feedback on whether we'd want to ask the maintainers to move the NodeProvider to another repo, cc @zhe-thoughts and @richardliaw.

In the meantime, the docs have contact links for the maintainers.

**Comment by DmitriGekhtman** [comment]
About K8s, I _think_ we've removed all docs that make explicit mention of using the cluster launcher for K8s.
We should make it clear if it's not already that K8s operators are the way to go.
@AmeerHajAli are there parts of the docs where this isn't clear?

File: doc/source/cluster/index.rst [hunk 1]
```diff
@@ -16,7 +16,7 @@ done so often), but the real power is using Ray on a cluster of machines.
 
 Ray can automatically interact with the cloud provider to request or release
 instances. You can specify :ref:`a configuration <cluster-config>` to launch
-clusters on :ref:`AWS, GCP, Azure, Kubernetes, Aliyun, on-premise, or even on
+clusters on :ref:`AWS, GCP, Azure (community managed), Kubernetes, on-premise, or even on
```
Comment (wuisawesome) [inline doc/source/cluster/index.rst:19]: Somewhat important nit: I think we should mention that either everythin (except k8s) is community supported, or not mention it, as it's implied by the fact that it's implied by the maintainer list on this page: https://docs.ray.io/en/latest/cluster/cloud.html

**Comment by DmitriGekhtman** [comment]
sry -- didn't realize this was a pull request.

@javi-redondo do you know if the page "cluster/index.rst" being edited is still accessible in the docs? I can't seem to find it with a web search. Was this orphaned with the new landing page?

**Comment by richardliaw** [comment]
It's ok to have community integrations and explicitly state that it is community maintained. The important thing is to set the expectations.

We have to let the community build things and participate in the project.

Thus let's keep this.

**Comment by DmitriGekhtman** [comment]
> It's ok to have community integrations and explicitly state that it is community maintained. The important thing is to set the expectations.
> 
> We have to let the community build things and participate in the project.
> 
> Thus let's keep this.

I think the call for now is to keep Aliyun and label both Aliyun and Azure as "community maintained."
It would be good either here or in a followup to explain that "community maintained" means "maintained by the authors, not the Ray team".

**Comment by DmitriGekhtman** [comment]
@AmeerHajAli thanks for bringing attention to how we communicate about community-maintained features!
I'll open a PR with doc changes reflecting the discussion.

**Comment by wuisawesome** [comment]
Closing in favor of https://github.com/ray-project/ray/pull/25687

**Pull request closed without merging**